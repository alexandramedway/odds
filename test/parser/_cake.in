do (() -> return 42)()
do print(
    ((x) -> return x + 1)(41)
)
