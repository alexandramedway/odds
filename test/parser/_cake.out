[
    Do(
        Cake(
            Fdecl({ params=[] ; body=[] ; return = Num_lit(42) }),
            []
        )
    ) ;
    Do(
        Call(Id(print),
        [
            Cake(
                Fdecl({
                    params=[x] ;
                    body=[] ;
                    return = Binop(Id(x), Add, Num_lit(1))
                }),
                [Num_lit(41)]
            )
        ])
    )
]
